### PR TITLE
[TextFields] New text field box style.

### PR DESCRIPTION
**Pull request by willlarche**
Dependent on #1816 , #1812 , #1814 , #1815 .

### CODE REVIEW

File: components/TextFields/src/MDCTextInputControllerTextFieldBox.m [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    textInsets.top = MDCTextInputTextFieldBoxNormalPlaceholderPadding;
+  }
+
+  // .bottom = underlineOffset + the half padding above the line but below the text field and any
+  // space needed for the labels and / or line.
+  // Legacy has an additional half padding here but this version does not.
+  CGFloat underlineOffset = [self underlineOffset];
+
+
+
```
Comment (ajsecord) [inline components/TextFields/src/MDCTextInputControllerTextFieldBox.m]: Delete extra blank lines.

File: components/TextFields/src/MDCTextInputControllerTextFieldBox.m [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                                                      toItem:self.textInput
+                                                   attribute:NSLayoutAttributeTop
+                                                  multiplier:1
+                                                    constant:underlineYConstant];
+    self.underlineY.active = YES;
+  }
+  self.underlineY.constant = underlineYConstant;
+}
+
+// Measurement from bottom to underline center Y
```
Comment (ajsecord) [inline components/TextFields/src/MDCTextInputControllerTextFieldBox.m]: Add period at end of sentence

File: components/TextFields/src/MDCTextInputControllerTextFieldBox.m [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  // The amount of space underneath the underline depends on whether there is content in the
+  // underline labels.
+  CGFloat underlineLabelsOffset = 0;
+  if (self.textInput.leadingUnderlineLabel.text.length) {
+    underlineLabelsOffset =
+    MDCCeil(self.textInput.leadingUnderlineLabel.font.lineHeight * 2.f) / 2.f;
+  }
+  if (self.textInput.trailingUnderlineLabel.text.length || self.characterCountMax) {
+    underlineLabelsOffset =
+    MAX(underlineLabelsOffset,
```
Comment (ajsecord) [inline components/TextFields/src/MDCTextInputControllerTextFieldBox.m]: +4

**Review by ajsecord — APPROVED** [review]
Small nits

File: components/TextFields/src/MDCTextInputControllerTextFieldBox.m (hunk 1 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerTextFieldBox.m]: Done!

File: components/TextFields/src/MDCTextInputControllerTextFieldBox.m (hunk 2 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerTextFieldBox.m]: Done!

File: components/TextFields/src/MDCTextInputControllerTextFieldBox.m (hunk 3 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerTextFieldBox.m]: Actually, I don't know why this is on its own line anyway. Fixed that and it's +4 beneath.

**Pull request merged**